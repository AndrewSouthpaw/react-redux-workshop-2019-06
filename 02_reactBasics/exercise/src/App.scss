body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
  "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
  sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
  monospace;
}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

img {
  height: 200px;
}

button {
  font-size: 26px;
}

.cage-display {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 300px;
}

.cage-img {
  position: relative;
}

.overlay {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  opacity: 0.5;
}

.overlay-yes {
  background-color: green;
}

.overlay-no {
  background-color: red;
}
