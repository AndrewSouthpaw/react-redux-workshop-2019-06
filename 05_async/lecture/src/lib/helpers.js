export const sleep = async timeInMs => new Promise(res => setTimeout(res, timeInMs))
