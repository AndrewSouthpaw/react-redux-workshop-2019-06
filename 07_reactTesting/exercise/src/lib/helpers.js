export const sleep = async timeInMs => new Promise(res => setTimeout(res, timeInMs))

export const fakeGetTodos = async () => {
  await sleep(1000)
  return { data: initialTodos }
}

export const fakeSaveTodoToServer = async (todo) => {
  await sleep(1000)
  return { data: todo }
}

export const initialTodos = [
  { id: 1, name: 'Learn about fetching data' },
  { id: 2, name: 'Discover async' },
  { id: 3, name: '???' },
  { id: 4, name: 'Profit' },
]

export const flushPromises = () => new Promise(resolve => setTimeout(resolve, 0))

export const reflush = async (wrapper) => {
  await flushPromises()
  wrapper.update()
}
