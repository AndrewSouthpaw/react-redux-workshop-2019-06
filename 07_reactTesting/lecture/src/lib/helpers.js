import { curry } from 'ramda'

export const sleep = async timeInMs => new Promise(res => setTimeout(res, timeInMs))

export const fakeGetTodos = async () => {
  await sleep(1000)
  return { data: initialTodos }
}

export const fakeSaveTodoToServer = async (todo) => {
  await sleep(1000)
  return { data: todo }
}

export const initialTodos = [
  { id: 1, name: 'Learn about fetching data' },
  { id: 2, name: 'Discover async' },
  { id: 3, name: '???' },
  { id: 4, name: 'Profit' },
]

export const flushPromises = () => new Promise(resolve => setTimeout(resolve, 0))

export const reflush = async (wrapper) => {
  await flushPromises()
  wrapper.update()
}

export const sel = curry((dataTestId, wrapper) => wrapper.find(`[data-test-id="${dataTestId}"]`))

/**
 * Workaround for broken `findWhere` behavior for mounted components when searching by text
 * https://github.com/airbnb/enzyme/issues/1566
 */
const textContent = (node) => {
  try {
    // enzyme sometimes blows up on text()
    return node.text()
  } catch (_e) {
    return ''
  }
}

// useful for finding a node by its text
export const findByText = (text, wrapper, options = {}) => {
  const comparator = options.exact ? x => textContent(x) === text : x => new RegExp(text).test(textContent(x))
  return wrapper.findWhere(comparator).filterWhere(x => x.html().startsWith('<')).last()
}
