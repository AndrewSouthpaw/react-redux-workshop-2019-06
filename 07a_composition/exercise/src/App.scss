body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
  "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
  sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
  monospace;
}

.App {
  text-align: center;
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.modal {
  color: black;
}

img {
  width: 100px;
}

.FancyBorder {
  width: 70%;
  padding: 10px 10px;
  border: 10px solid;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.FancyBorder-blue {
  border-color: blue;
}

.FancyBorder-blue {
  border-color: blue;
}

.Dialog-title {
  margin: 0;
  font-family: sans-serif;
}

.Dialog-message {
  font-size: larger;
}

#counter {
  border: 2px solid #cc0000;
  color: #ffffff;
  text-align: center;
  padding: 4px;
  width: 150px;
  background: #000000;
  font: normal 12px tahoma, verdana, Arial;
}

#counter span {
  font-weight: bold;
}
