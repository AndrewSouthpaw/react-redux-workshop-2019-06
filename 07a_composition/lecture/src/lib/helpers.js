import React from 'react'
import { mount } from 'enzyme'
import { curry } from 'ramda'

export const sleep = async timeInMs => new Promise(res => setTimeout(res, timeInMs))

export const setupMount = (Component, defaultPropsFn) => (props) => (
  mount(<Component {...defaultPropsFn()} {...props} />)
)

export const sel = curry((dataTestId, wrapper) => wrapper.find(`[data-test-id="${dataTestId}"]`))

/**
 * Workaround for broken `findWhere` behavior for mounted components when searching by text
 * https://github.com/airbnb/enzyme/issues/1566
 */
const textContent = (node) => {
  try {
    // enzyme sometimes blows up on text()
    return node.text()
  } catch (_e) {
    return ''
  }
}

// useful for finding a node by its text
export const findByText = (text, wrapper, options = {}) => {
  const comparator = options.exact ? x => textContent(x) === text : x => new RegExp(text).test(textContent(x))
  return wrapper.findWhere(comparator).filterWhere(x => x.html().startsWith('<')).last()
}
