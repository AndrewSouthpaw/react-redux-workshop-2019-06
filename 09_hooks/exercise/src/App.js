import React from 'react'
import { PizzaTracker } from './PizzaTracker'

export class App extends React.Component {
  static initialOrders = [
    { id: 0, customerName: 'Steve', progress: 0 },
    { id: 1, customerName: 'Mary', progress: 1 },
    { id: 2, customerName: 'Tom', progress: 2 },
    { id: 3, customerName: 'Sue', progress: 3 },
    { id: 4, customerName: 'David', progress: 4 },
  ]

  state = {
    orders: App.initialOrders,
  }

  updateProgress = (orderId, newProgress) => {
    let newOrders = []
    this.state.orders.forEach((value) => {
      if (value.id === orderId) value.progress = newProgress
      newOrders.push(value)
    })
    this.setState({ orders: newOrders })
  }

  completeOrder = (orderId) => {
    let newOrders = []
    this.state.orders.forEach((value) => {
      if (value.id === orderId) return
      newOrders.push(value)
    })
    this.setState({ orders: newOrders })
  }

  render() {
    const { orders } = this.state
    return (
      <div className="App">
        {orders.map((value) => {
          return <PizzaTracker
            progress={value.progress} customerName={value.customerName}
            orderReceived={value.orderReceived}
            id={value.id} key={value.id}
            onUpdateProgress={this.updateProgress}
            onCompleteOrder={this.completeOrder}
          />
        })}
      </div>
    )
  }
}
