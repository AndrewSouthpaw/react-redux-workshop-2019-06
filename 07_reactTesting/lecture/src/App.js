import React from 'react'
import avatar from './avatar.jpg'
import './App.scss'

export class NameBadge extends React.Component {
  state = {
    following: false,
  }

  addFollower = () => {
    this.setState({ following: !this.state.following })
  }

  render() {
    const { name, avatar } = this.props
    const { following } = this.state
    return (
      <div className="NameBadge">
        <div>
          <img className="avatar" src={avatar} />
        </div>
        <div>
          <p>
            {name}
            {following && ' (Following)'}
          </p>
          <button onClick={this.addFollower}>{following ? 'Unfollow' : 'Follow'}</button>
        </div>
      </div>
    )
  }
}

export class App extends React.Component {
  render() {
    return (<NameBadge name="Andrew Smith" avatar={avatar} />)
  }
}
