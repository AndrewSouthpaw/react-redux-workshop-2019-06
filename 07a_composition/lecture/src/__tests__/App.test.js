import React from 'react'
import { findByText, sel, setupMount } from '../lib/helpers'
import { App } from '../App.begin'
import {shallow} from 'enzyme'

const defaultProps = () => ({})

const wrap = setupMount(App, defaultProps)

describe('App', () => {
  it('shallow can be deceptive or useless...', () => {
    const w = shallow(<App />)
    expect(w.debug()).toMatchSnapshot()
  })

  it('should show some text', () => {
    const w = wrap()
    expect(w.text()).toEqual(
      'Welcome' +
      'Thank you for visiting us on the interwebs! ' +
      'Favorite links:CSS Puns' +
      'Guestbook' +
      'Please sign my guestbook!' +
      'There are' +
      '2,389,987' +
      'hits this month',
    )
  })
})
