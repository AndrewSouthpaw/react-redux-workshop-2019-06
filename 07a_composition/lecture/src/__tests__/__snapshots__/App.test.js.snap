// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App shallow can be deceptive or useless... 1`] = `
"<div className=\\"App\\">
  <WelcomeDialog />
  <Guestbook />
  <Counter />
</div>"
`;
